src/modules/rating/controllers/rating.controller.ts

```diff
@@ -0,0 +1,39 @@
+import { Body, Controller, Logger, Param, Post } from '@nestjs/common'
+import { RatingRepository } from '../database/rating.repository'
+import {
+	CreateRatingRequestDTO,
+	CreateRatingResponseDTO,
+} from './dtos/create-rating.dtos'
+import { ApiResponse, ApiTags } from '@nestjs/swagger'
+import { Rating } from '../database/rating.model'
+import { ObjectIdParam } from '@modules/admin/product/controllers/dtos/common.dto'
+
+@Controller('v1/products')
+@ApiTags('Rating')
+export class RatingController {
+	private readonly logger: Logger = new Logger(RatingController.name)
+	constructor(private readonly ratingRepo: RatingRepository) {}
+	@Post('/:id/ratings')
+	@ApiResponse({
+		status: 201,
+		type: CreateRatingRequestDTO,
+	})
+	async create(
+		@Param() { id }: ObjectIdParam,
+		@Body() dto: CreateRatingRequestDTO,
+	): Promise<CreateRatingResponseDTO> {
+		const ratingId = this.ratingRepo.genId()
+		const rating: Rating = {
+			...dto,
+			_id: ratingId,
+			product_id: id,
+			status: 'Pending',
+		}
+		try {
+			const result = await this.ratingRepo.createRating(rating)
+			return new CreateRatingResponseDTO({ data: result })
+		} catch (error) {
+			this.logger.error(error)
```

Comment:
Should throw error if error occurs. In this case, the error should be thrown is : BadRequestException